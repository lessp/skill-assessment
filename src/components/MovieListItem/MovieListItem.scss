.App-movieListItem {
  flex-basis: 50%;
  max-width: 25.6rem;

  display: flex;
  flex-direction: column;
  justify-content: center;

  padding: 1.2rem;
  cursor: pointer;

  transition: all 300ms ease;

  figure {
    img {
      display: block;
      width: 100%;
      height: auto;

      box-shadow: 0 14px 32px 0 rgba(0, 0, 0, 0.15);
      border-radius: 5px;
      transition: all 300ms ease;

      margin-bottom: 0.8rem;
    }

    figcaption {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      text-align: center;

      font-weight: 600;
      color: var(--color-default);

      span {
        margin-top: 0.4rem;
        opacity: 0.72;
        font-weight: 500;
      }
    }
  }

  &:hover {
    transform: scale(1.05);
  }
}

@media screen and (min-width: 768px) {
  .App-movieListItem {
    flex-basis: 25%;
    padding: 2.4rem;

    figure {
      figcaption {
        font-size: 1.6rem;

        span {
          font-size: 1.4rem;
        }
      }
    }
  }
}
