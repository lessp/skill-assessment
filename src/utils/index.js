export const TMDB_BASE_URL = 'https://api.themoviedb.org/3';

export const TMDB_SEARCH_MOVIES_URL = `${TMDB_BASE_URL}/search/movie?api_key=${
  process.env.REACT_APP_TMDB_API_KEY
}&language=en-US&page=1&include_adult=false`;

export const movieDetailsUrl = movieId =>
  `${TMDB_BASE_URL}/movie/${movieId}/credits?api_key=${
    process.env.REACT_APP_TMDB_API_KEY
  }`;

export const actorDetailsUrl = personId =>
  `https://api.themoviedb.org/3/person/${personId}?api_key=${
    process.env.REACT_APP_TMDB_API_KEY
  }&language=en-US`;

export const actorCreditsUrl = personId =>
  `https://api.themoviedb.org/3/person/${personId}/movie_credits?api_key=${
    process.env.REACT_APP_TMDB_API_KEY
  }&language=en-US`;

export const TMDB_IMAGES_URL = `https://image.tmdb.org/t/p`;

export const normaliseMovies = movies =>
  movies.map(movie => {
    movie.poster_path = TMDB_IMAGES_URL + `/w500/${movie.poster_path}`;
    movie.backdrop_path = TMDB_IMAGES_URL + `/w1280/${movie.backdrop_path}`;
    return movie;
  });

export const normaliseCrew = crew =>
  crew.reduce((jobs, member) => {
    const crewPosition = member.job.toLowerCase();

    jobs[crewPosition] = jobs[crewPosition] || [];
    jobs[crewPosition].push(member);

    return jobs;
  }, {});

export const normaliseCast = cast =>
  cast.map(actor => {
    actor.profile_path = TMDB_IMAGES_URL + `/w300/${actor.profile_path}`;
    return actor;
  });

export const normaliseActor = actor => ({
  ...actor,
  profile_path: TMDB_IMAGES_URL + `/w300/${actor.profile_path}`
});

export const sortByPopularity = credits =>
  credits.slice().sort((x, y) => y.popularity - x.popularity);
