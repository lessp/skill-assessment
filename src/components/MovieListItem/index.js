import React from 'react';

import './MovieListItem.css';

const MovieListItem = ({ movie, ...props }) => (
  <li className="App-movieListItem" {...props}>
    <figure>
      <img src={movie.poster_path} alt={movie.title} />
      <figcaption>
        {movie.title} <span>({movie.release_date.substring(0, 4)})</span>
      </figcaption>
    </figure>
  </li>
);

export default MovieListItem;
