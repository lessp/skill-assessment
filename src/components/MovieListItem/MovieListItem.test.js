import React from 'react';
import MovieListItem from './';
import renderer from 'react-test-renderer';

import { TMDB_SEARCH_MOVIES_URL } from '../../utils';

it('renders MovieListItem with dummy API-data', () => {
  const movie = {
    poster_path: 'dummyImageUrl.png',
    title: 'Movie Title',
    release_date: '1970-01-01'
  };
  const tree = renderer.create(<MovieListItem movie={movie} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders MovieListItem with real API-data', async () => {
  const movie = await fetch(
    `${TMDB_SEARCH_MOVIES_URL}&query=mission%20impossible`
  ).then(res => res.json());

  const tree = renderer
    .create(<MovieListItem movie={movie.results[0]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
