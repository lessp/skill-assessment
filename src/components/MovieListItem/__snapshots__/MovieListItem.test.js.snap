// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieListItem with dummy API-data 1`] = `
<li
  className="App-movieListItem"
>
  <figure>
    <img
      alt="Movie Title"
      src="dummyImageUrl.png"
    />
    <figcaption>
      Movie Title
       
      <span>
        (
        1970
        )
      </span>
    </figcaption>
  </figure>
</li>
`;

exports[`renders MovieListItem with real API-data 1`] = `
<li
  className="App-movieListItem"
>
  <figure>
    <img
      alt="Mission: Impossible"
      src="/vmj2PzTLC6xJvshpq8SlaYE3gbd.jpg"
    />
    <figcaption>
      Mission: Impossible
       
      <span>
        (
        1996
        )
      </span>
    </figcaption>
  </figure>
</li>
`;
